src/vmm/src/persist.rs

```diff
@@ -502,6 +525,47 @@ mod tests {
         )
     }
 
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn test_get_snapshot_data_version() {
+        let mmio_base = 0xd000_0000;
+        let device_manager = MMIODeviceManager::new(mmio_base, (arch::IRQ_BASE, arch::IRQ_MAX));
+
+        assert!(get_snapshot_data_version(&None, &VERSION_MAP, &device_manager).is_ok());
+        // Validate sanity checks fail because of invalid target version.
+        assert!(get_snapshot_data_version(
+            &Some(String::from("foo")),
+            &VERSION_MAP,
+            &device_manager
+        )
+        .is_err());
+
+        for version in FC_VERSION_TO_SNAP_VERSION.keys() {
+            let res =
+                get_snapshot_data_version(&Some(version.to_owned()), &VERSION_MAP, &device_manager);
+
+            assert!(res.is_ok());
+        }
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    #[test]
+    fn test_get_snapshot_data_version() {
```

Comment:
After the use of vmm as a parameter, this function can easily be melded into the x86_64 specific one.